### ISSUE TITLE
Unexpected ref safety error assigning ref of captured local

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#65353]
An error is reported for the ref assignment in the more nested lambda expression:
```csharp
delegate void D();

class Program
{
    static void F(D d) { }

    static void Main()
    {
        {
            int x = 1;
            F(() =>
                {
                    int y = 2;
                    ref int r = ref y;
                    r = ref x;
                });
        }
        {{{
            int x = 1;
            F(() =>
                {
                    int y = 2;
                    ref int r = ref y;
                    r = ref x; // error CS8374: ... 'x' has a narrower escape scope than 'r'
                });
        }}}
    }
}
```
Result:
```
(24,21): error CS8374: Cannot ref-assign 'x' to 'r' because 'x' has a narrower escape scope than 'r'.
```

This is not a recent regression. The same error is reported in csc 3.11.

**Comment by cston** [comment]
The same issue exists with local functions:
```csharp
#pragma warning disable 8321

class Program
{
    static void Main()
    {
        {{{
        int x = 1;
        void F()
        {
            int y = 2;
            ref int r = ref y;
            r = ref x; // error CS8374: ... 'x' has a narrower escape scope than 'r'
        }
        }}}
    }
}
```
